tools/docker/compose.md

```diff
@@ -0,0 +1,145 @@
+
+```
+FROM fedora
+MAINTAINER http://fedoraproject.org/wiki/Cloud
+
+RUN dnf -y update && dnf clean all
+RUN dnf -y install python-pip python-django git sqlite python-psycopg2 && dnf clean all
+
+# create directory /code and mount sources there
+RUN mkdir /code
+WORKDIR /code
+VOLUME /code
+
+EXPOSE 8000
+
+CMD [ "./manage.py runserver 0.0.0.0:8000" ]
+```
+
+```
+$ cd Fedora-Dockerfiles/Django
+$ docker build --tag=fedora-django .
+$ cd ../..
+```
+
+And now PostgreSQL:
+
+```
+$ cd Fedora-Dockerfiles/postgresql
+$ docker build --tag=fedora-postgresql .
+```
+
+Don't forget to install django on your workstation, so you have access to `django-admin` command.
+
+
+## Create django project
+
+Let's start new project with command `django-admin`:
+
+```
+$ django-admin startproject awesome_web
+$ cd awesome_web
+```
+
+Since we use PostgreSQL, we need change `DATABASE` variable in `settings.py`:
+```
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'awesome_web',
+        'USER': 'awesome_web_user',
+        'PASSWORD': 'secret_password',
+        'HOST': 'db',
+        'PORT': 5432,
+    }
+}
+```
+
+## Let's compose the development environment now
+
+Config for docker-compose is fairly easy:
+
+```
+$ cat docker-compose.yml
+web:
+  image: fedora-django
+  ports:
+   - "8000:8000"
+  volumes:
+   - .:/code
+  links:
+   - db
+db:
+  image: fedora-postgresql
+  volumes:
+   - $YOUR_DIR:/var/lib/pgsql/data
+  environment:
+   - POSTGRESQL_DATABASE=awesome_web
+   - POSTGRESQL_USER=awesome_web_user
+   - POSTGRESQL_PASSWORD=secret_password
+```
+
+So, we need to populate database which is done by `manage.py migrate`. We need to run the database container first, then populate the database:
+
+```
+$ docker-compose up -d db
+ docker-compose run web python manage.py migrate
+Operations to perform:
+  Synchronize unmigrated apps: staticfiles, messages
+  Apply all migrations: admin, contenttypes, auth, sessions
+Synchronizing apps without migrations:
+  Creating tables...
+    Running deferred SQL...
+  Installing custom SQL...
+Running migrations:
+  Rendering model states... DONE
+  Applying contenttypes.0001_initial... OK
+  Applying auth.0001_initial... OK
+  Applying admin.0001_initial... OK
+  Applying contenttypes.0002_remove_content_type_name... OK
+  Applying auth.0002_alter_permission_name_max_length... OK
+  Applying auth.0003_alter_user_email_max_length... OK
+  Applying auth.0004_alter_user_username_opts... OK
+  Applying auth.0005_alter_user_last_login_null... OK
+  Applying auth.0006_require_contenttypes_0002... OK
+  Applying sessions.0001_initial... OK
+
+```
+
+Time to spin the whole environment!
+
+```
+$ docker-compose up
+```
+
+That's it! You have your development environment running in docker containers! You can check easily:
+
+```
+elinks -dump http://$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' awesomeweb_web_1):8000
```

Comment:
Yes, thanks. Will definitely read the whole thing several times before it's done and ready for merge -- this is basically still a wip.